OFBIZ-13313

gtchaboussie opened the pull request:
EntityQuery use dispatcher

Improved: Creates an interface for queryable objects, or objects that have a delegator.
(OFBIZ-13313)

Explanation:  Allows the use of the dispatcher directly in the EntityQuery.
Created the Queryable interface to do so.

Thanks: to Jacopo and Nicolas for the feedback.

Commit:
OFBIZ-13313
Creates an interface for queryable objects, or objects that have a delegator.
Allows for shorter queries.
Thanks to Jacopo and Nicolas for the feedback.

Commit:
OFBIZ-13313
Fixes the param name in javadoc

jacopoc commented on framework/entity/src/main/java/org/apache/ofbiz/entity/Delegator.java:
```diff
@@ -43,15 +43,18 @@
 
     enum OperationType { INSERT, UPDATE, DELETE }
 
+    Delegator getDelegator();
```
This declaration is not required since it is an inherited method.

jacopoc commented on framework/entity/src/main/java/org/apache/ofbiz/entity/util/Queryable.java:
```diff
@@ -0,0 +1,17 @@
+import org.apache.ofbiz.entity.Delegator;
+
+/**
+ * Interface used for the <code>EntityQuery</code> to set the delegator to yse for the query.
```
The Javadoc should just describe the capability of the interface without directly mentioning the classes that may use it (such as EntityQuery). There are also a few typos, such as: "yse", "easyer".

jacopoc commented on framework/entity/src/main/java/org/apache/ofbiz/entity/util/Queryable.java:
```diff
@@ -0,0 +1,17 @@
+public interface Queryable {
+
+    /**
+     * Gets the GenericEntityDelegator associated with this dispatcher
```
The Javadoc is not correct as it mentions GenericEntityDelegator rather than Delegator and it mentions "this dispatcher" which is out of context in this interface.

jacopoc commented:
Thanks Gaetan,

The introduction of the new interface makes sense and solves the concern I had.
However in my opinion the name "Queryable" doesn't describe very well the capability of the new interface. A better name could be "DelegatorProvider".
I have also added a few comments inline.

Commit:
OFBIZ-13313
Fixes points raised by Jacopo.
Thanks for the review.

Commit:
OFBIZ-13313
Typo

gtchaboussie commented:
Thanks for the review, all issues should have been fixed.
The name of "DelegatorProvider" sounds fine to me, and more descriptive of the intention of the interface.

jacopoc commented:
Thanks Gaetan, it looks good to me now.

nmalin reviewed (changes requested):
Hello Gaëtan,

I added some improvement on the following patch.
* Add Licence Apache 2.0 on header for DelegatorProvider
* GenericEntity implements DelegatorProvider
* Add test on EntityQuery to be sure that the DelegatorProvider and EntityQuery.use(DelegatorFactory) work well

Thanks for your work !

[OFBiz-PR923.patch](https://github.com/user-attachments/files/24879276/OFBiz-PR923.patch)

Commit:
Improved : Added an interface for all objects that holds delegator (OFBIZ-13313)
- Creates an interface for queryable objects, or objects that have a delegator.
- Allows for shorter queries.
Thanks to Jacopo and Nicolas for the feedback and review
